recipe/meta.yaml

```
@@ -1,108 +1,113 @@
 {% set name = "scikit-learn" %}
-{% set version = "1.6.1" %}
+{% set version = "1.7.0" %}
+{% set sha256 = "c01e869b15aec88e2cdb73d27f15bdbe03bce8e2fb43afbe77c45d399e73a5a3" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/scikit_learn-{{ version }}.tar.gz
-  sha256: b4fc2525eca2c69a59260f583c56a7557c6ccdf8deafdba6e060f94c1c59738e
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name.replace('-', '_')}}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
 
 build:
   number: 0
-  skip: True  # [py<39]
-  script:                                                                                  # [not win]
+  script_env:  # [win]
+    - SCIKIT_LEARN_SHA256={{ sha256 }}  # [win]
+  script:  # [not win]
     - {{ PYTHON }} -m pip install --no-deps --ignore-installed --no-build-isolation . -vv  # [not win]
   missing_dso_whitelist:
     - '*/libc++.1.dylib'  # [osx]
-    - '*/libomp.dylib'    # [osx]
+    - '*/libomp.dylib'  # [osx]
+  skip: True  # [py<310]
+  #ignore_run_exports:
+  #  - numpy
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
-    # Host dependencies are pinned to last build
+    - pip
     - python
-    - cython >=3.0.10
-    - numpy 2.0  # [py<313]
-    - numpy 2.1  # [py>=313]
+    - cython >=3.0.10,<3.2.0
+    - numpy {{ numpy }}
     - llvm-openmp 14.0.6  # [osx]
-    - pip
-    - scipy >=1.6.0
-    - meson-python >=0.16.0
-    - python-build        # [win]
-    # mkl variant pulls in intel-openmp which conflicts with
-    # llvm-openmp. i.e. force to use openblas variant of numpy, scipy,
-    # etc.
-    - blas * openblas     # [osx and x86_64]
+    - scipy >=1.8.0,<1.16.0
+    - meson-python >=0.16.0,<0.19.0
+    - python-build  # [win]
   run:
     - python
     - joblib >=1.2.0
-    - numpy >=1.19.5
-    - scipy >=1.6.0      # [not (osx and x86_64)]
+    - numpy >=1.22.0
+    - scipy >=1.8.0      # [not (osx and x86_64)]
     # Build fails on osx-64 without the newest version of scipy.
     # https://github.com/scikit-learn/scikit-learn/issues/28367#issuecomment-1941101095
-    - scipy >=1.13.0      # [osx and x86_64]
+    - scipy >=1.13.0,<1.16.0  # [osx and x86_64]
```

```suggestion
    - scipy >=1.13.0 # [osx and x86_64]
```
I don't see upper bounds there: https://github.com/scikit-learn/scikit-learn/blob/1.7.1/pyproject.toml#L11